Zeplin.sketchplugin/Contents/Sketch/utils.cocoascript

```diff
@@ -1,35 +1,52 @@
-var documentExportable = function (context) {
-    var doc = context.document;
+var temporaryPath = function() {
+    var name = [[[NSUUID UUID] UUIDString] stringByAppendingPathExtension:@"sketch"];
+    var temporaryDirectory = NSTemporaryDirectory();
+    var temporaryZeplinDirectory = [temporaryDirectory stringByAppendingPathComponent:@"io.zeplin.osx"];
     
-    if (![doc fileURL] || [doc isDraft]) {
-        [NSApp displayDialog:@"Please save the document before exporting to Zeplin." withTitle:@"Document not saved"];
+    var isDir = MOPointer.alloc().initWithValue_(false);
+    var fileManager = [NSFileManager defaultManager];
+    if (![fileManager fileExistsAtPath:temporaryZeplinDirectory isDirectory:isDir] || isDir.value() == false) {
+        [fileManager removeItemAtPath:temporaryZeplinDirectory error:nil];
+        [fileManager createDirectoryAtPath:temporaryZeplinDirectory withIntermediateDirectories:true attributes:nil error:nil];
+    }
+
+    var path = [temporaryZeplinDirectory stringByAppendingPathComponent:name];
+
+    name = nil;
+    temporaryDirectory = nil;
+    temporaryZeplinDirectory = nil;
+    isDir = nil;
+    fileManager = nil;
 
-        return false;
+    return path;
+}
+
+var temporaryPathForDesignFile = function (context) {
+    var doc = context.document;
+    var path = temporaryPath();
+
+    [doc showMessage:@"Saving document…"];
+    [doc saveToURL:[NSURL fileURLWithPath: path] ofType:@"com.bohemiancoding.sketch.drawing.single" forSaveOperation:NSSaveToOperation delegate:nil didSaveSelector:nil contextInfo:nil];
+
+    var fileManager = [NSFileManager defaultManager];
+    var stopTime = [NSDate dateWithTimeIntervalSinceNow:30];
```

Comment:
wdyt about this interval value?

Reply:
I think 30 secs should be more than enough. 👍